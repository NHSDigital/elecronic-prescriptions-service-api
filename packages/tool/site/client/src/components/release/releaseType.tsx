import * as React from "react"
import {TextInput, Textarea} from "nhsuk-react-components"
import {Field} from "formik"
import RadioField from "../common/radioField"

interface ReleaseTypeProps {
  initialValue: string
  value: string
  error?: string
}

const ReleaseType: React.FC<ReleaseTypeProps> = ({
  initialValue,
  value,
  error
}) => {
  return (
    <>
      <RadioField
        name="releaseType"
        label="Choose how you want to release prescription(s)"
        defaultValue={initialValue}
        error={error}
        fieldRadios={[
          {
            id: 0,
            value: "all",
            text: "All nominated prescriptions for the below pharmacy"
          },
          {
            id: 1,
            value: "prescriptionId",
            text: "A single prescription by ID"
          },
          {
            id: 2,
            value: "custom",
            text: "With a FHIR release message"
          }
        ]}
      />
      {value === "prescriptionId" &&
        <Field
          id="prescriptionId"
          name="prescriptionId"
          as={TextInput}
          width={30}
          label="Prescription ID"
        />
      }
      {value === "custom" &&
        <Field
          id="customReleaseFhir"
          name="customReleaseFhir"
          as={Textarea}
          rows={20}
          label="Paste a FHIR release message"
        />
      }
    </>
  )
}

export default ReleaseType
