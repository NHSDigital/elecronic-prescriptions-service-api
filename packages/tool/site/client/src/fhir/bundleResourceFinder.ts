import * as fhir from "fhir/r4"
import {MedicationDispense, MedicationRequest} from "./helpers"

export const getPatientResources = buildResourceFinder<fhir.Patient>("Patient")
export const getPractitionerResources = buildResourceFinder<fhir.Practitioner>("Practitioner")
export const getPractitionerRoleResources = buildResourceFinder<fhir.PractitionerRole>("PractitionerRole")
export const getOrganizationResources = buildResourceFinder<fhir.Organization>("Organization")
export const getMedicationRequestResources = buildResourceFinder<MedicationRequest>("MedicationRequest")
export const getMedicationDispenseResources = buildResourceFinder<MedicationDispense>("MedicationDispense")
export const getMessageHeaderResources = buildResourceFinder<fhir.MessageHeader>("MessageHeader")
export const getCommunicationRequestResources = buildResourceFinder<fhir.CommunicationRequest>("CommunicationRequest")

function buildResourceFinder<T extends fhir.FhirResource>(resourceType: string): (payload: fhir.Bundle) => Array<T> {
  return function (bundle: fhir.Bundle): Array<T> {
    const typeCheck = (resource: fhir.FhirResource): resource is T => resource.resourceType === resourceType
    return bundle.entry.map(entry => entry.resource).filter(typeCheck)
  }
}
