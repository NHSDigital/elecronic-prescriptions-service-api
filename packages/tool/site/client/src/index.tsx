import * as React from "react"
import {Page} from "./components/page"
import SignPage from "./pages/signPage"
import * as ReactDOM from "react-dom"
import {BrowserRouter, Route, Switch} from "react-router-dom"
import ClaimPage from "./pages/claimPage"
import DispensePage from "./pages/dispensePage"
import {CookiesProvider} from "react-cookie"
import SendPage from "./pages/sendPage"
import HomePage from "./pages/homePage"
import PrescriptionSearchPage from "./pages/prescriptionSearchPage"
import ReleasePage from "./pages/releasePage"
import CancelPage from "./pages/cancelPage"
import MyPrescriptionsPage from "./pages/myPrescriptionsPage"
import LoginPage from "./pages/loginPage"
import ValidatePage from "./pages/validatePage"
import ReturnPage from "./pages/returnPage"
import WithdrawPage from "./pages/withdrawPage"
import LoadPage from "./pages/loadPage"
import {Environment} from "./services/environment"
import LogoutPage from "./pages/logoutPage"
import ConfigPage from "./pages/configPage"
import ComparePage from "./pages/comparePage"
import ViewPrescriptionPage from "./pages/viewPrescriptionPage"
import {PageFullWidth} from "./components/pageFullWidth"
import {PageContainer} from "./components/pageContainer"
import DoseToTextPage from "./pages/doseToTextPage"
import TrackerViewPrescriptionPage from "./pages/trackerViewPrescription"

const customWindow = window as Record<string, any>

export interface AppContextValue {
  baseUrl: string
  environment: Environment
}

export const AppContext = React.createContext<AppContextValue>({baseUrl: "/", environment: "int"})

async function startApplication(baseUrl: string, environment: Environment): Promise<void> {
  const urlParams = new URLSearchParams(window.location.search)
  const content = (
    <AppContext.Provider value={{baseUrl, environment}}>
      <CookiesProvider>
        <Page>
          <BrowserRouter>
            <Switch>
              <Route path={`${baseUrl}/`}>
                <PageContainer>
                  <HomePage />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}dose-to-text`}>
                <PageContainer>
                  <DoseToTextPage/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}config`}>
                <PageContainer>
                  <ConfigPage/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}my-prescriptions`}>
                <PageContainer>
                  <MyPrescriptionsPage/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}login`}>
                <PageContainer>
                  <LoginPage separateAuth={urlParams.get("separate_auth")}/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}logout`}>
                <PageContainer>
                  <LogoutPage/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}prescribe/load`}>
                <PageContainer>
                  <LoadPage/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}prescribe/edit`}>
                <PageContainer>
                  <SignPage/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}prescribe/send`}>
                <PageContainer>
                  <SendPage token={urlParams.get("token")} state={urlParams.get("state")}/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}prescribe/cancel`}>
                <PageContainer>
                  <CancelPage prescriptionId={urlParams.get("prescription_id")}/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}dispense/release`}>
                <PageContainer>
                  <ReleasePage prescriptionId={urlParams.get("prescription_id")}/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}dispense/return`}>
                <PageContainer>
                  <ReturnPage prescriptionId={urlParams.get("prescription_id")} />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}dispense/withdraw`}>
                <PageContainer>
                  <WithdrawPage prescriptionId={urlParams.get("prescription_id")} />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}dispense/dispense`}>
                <PageContainer>
                  <DispensePage prescriptionId={urlParams.get("prescription_id")} amendId={urlParams.get("amend_id")} />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}dispense/claim`}>
                <PageContainer>
                  <ClaimPage prescriptionId={urlParams.get("prescription_id")} amend={!!urlParams.get("amend")}/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}search`}>
                <PageContainer>
                  <PrescriptionSearchPage prescriptionId={urlParams.get("prescription_id")} />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}tracker`}>
                <PageContainer>
                  <TrackerViewPrescriptionPage prescriptionId={urlParams.get("prescription_id")} />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}view`}>
                <PageContainer>
                  <ViewPrescriptionPage prescriptionId={urlParams.get("prescription_id")} />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}validate`}>
                <PageContainer>
                  <ValidatePage />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}compare-prescriptions`}>
                <PageFullWidth>
                  <ComparePage />
                </PageFullWidth>
              </Route>
            </Switch>
          </BrowserRouter>
        </Page>
      </CookiesProvider>
    </AppContext.Provider>
  )
  ReactDOM.render(content, document.getElementById("root"))
}

customWindow.startApplication = startApplication
