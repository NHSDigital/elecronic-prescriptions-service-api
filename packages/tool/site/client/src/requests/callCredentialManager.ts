import {HubResponse, start} from "./helpers"

// eslint-disable-next-line @typescript-eslint/no-explicit-any
(window as any).jQuery = jQuery;
// eslint-disable-next-line @typescript-eslint/no-explicit-any
(window as any).$ = jQuery

const SIGNALR_URL = `"http://localhost:"+prService.portNumber()+"/signalr"`
const SIGNALR_HUB_NAME = "signingHub"
const SIGNALR_METHOD_NAME = "requestToSign"

export async function sign(jwt: string): Promise<HubResponse> {
  const connection = $.hubConnection(SIGNALR_URL, {logging: true})
  if (connection.state !== SignalR.ConnectionState.Disconnected) {
    console.log(`Unexpected SignalR connection state ${connection.state}`)
    throw new Error("SignalR connection is already active.")
  }

  try {
    const proxy = connection.createHubProxy(SIGNALR_HUB_NAME)
    await connection.start({transport: "longPolling"})
    return await proxy.invoke(SIGNALR_METHOD_NAME, jwt)
  } finally {
    connection.stop()
  }
}

window.addEventListener("DOMContentLoaded", () => {
  start(sign)
})
