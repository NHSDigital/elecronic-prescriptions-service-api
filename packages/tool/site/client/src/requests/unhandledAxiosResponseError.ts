import {AxiosResponse} from "axios"

export class UnhandledAxiosResponseError extends Error {
  response: AxiosResponse

  constructor(message: string, response: AxiosResponse) {
    super(message)
    this.response = response
  }
}
