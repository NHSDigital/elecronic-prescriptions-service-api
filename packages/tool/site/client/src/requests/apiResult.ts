import {FhirResource} from "fhir/r4"

export function isApiResult(data: unknown): data is ApiResult {
  const result = data as ApiResult
  return typeof result.success === "boolean"
    && "request" in result
    && "response" in result
}

export interface ApiResult {
  success: boolean
  request: FhirResource
  request_xml?: string
  response: FhirResource
  response_xml?: string
}
