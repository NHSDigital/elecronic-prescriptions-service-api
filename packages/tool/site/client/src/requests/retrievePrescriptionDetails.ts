/* eslint-disable max-len */
import {axiosInstance} from "./axiosInstance"
import * as fhir from "fhir/r4"
import {getResponseDataIfValid} from "./getValidResponse"
import {getArrayTypeGuard, isBundle} from "../fhir/typeGuards"

export async function getPrescriptionOrderMessage(baseUrl: string, prescriptionId: string): Promise<fhir.Bundle> {
  const prescriptionOrderResponse = await axiosInstance.get<fhir.Bundle>(`${baseUrl}dispense/release/${prescriptionId}`)
  return getResponseDataIfValid(prescriptionOrderResponse, isBundle)
}

export async function getDispenseNotificationMessages(baseUrl: string, prescriptionId: string): Promise<Array<fhir.Bundle>> {
  const dispenseNotificationsResponse = await axiosInstance.get<Array<fhir.Bundle>>(`${baseUrl}dispenseNotifications/${prescriptionId}`)
  return getResponseDataIfValid(dispenseNotificationsResponse, getArrayTypeGuard(isBundle))
}
