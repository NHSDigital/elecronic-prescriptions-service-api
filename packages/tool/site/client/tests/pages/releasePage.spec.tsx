import {waitFor} from "@testing-library/react"
import {screen} from "@testing-library/dom"
import pretty from "pretty"
import * as React from "react"
import MockAdapter from "axios-mock-adapter"
import {AppContextValue} from "../../src"
import {renderWithContext} from "../renderWithContext"
import ReleasePage, {DispenserDetails, createRelease} from "../../src/pages/releasePage"
import userEvent from "@testing-library/user-event"
import {axiosInstance} from "../../src/requests/axiosInstance"
import {internalDev} from "../../src/services/environment"
import {ReleaseFormValues} from "../../src/components/release/releaseForm"
import * as fhir from "fhir/r4"
import {BrowserRouter} from "react-router-dom"

const baseUrl = "baseUrl/"
const prescriptionId = "7A9089-A83008-56A03J"
const context: AppContextValue = {baseUrl, environment: internalDev}

const releaseUrl = `${baseUrl}dispense/release`

const mock = new MockAdapter(axiosInstance)

beforeEach(() => mock.reset())
afterEach(() => mock.reset())

test("Displays release form", async () => {
  const container = await renderPage()

  expect(screen.getByText("Release prescription(s)")).toBeTruthy()
  expect(pretty(container.innerHTML)).toMatchSnapshot()
})

test("Displays release result", async () => {
  mock.onAny(releaseUrl).reply(200, {
    prescriptionIds: [],
    success: true,
    request: {req: "JSON Request"},
    request_xml: "XML Request",
    response: {res: "JSON Response"},
    response_xml: "XML Response"
  })

  const container = await renderPage()
  const pharmacyContainer = await screen.findByLabelText<HTMLElement>("Pharmacy to release prescriptions to")
  const pharmacyRadios = pharmacyContainer.getElementsByClassName("nhsuk-radios__input") as HTMLCollectionOf<HTMLInputElement>
  userEvent.click(pharmacyRadios[0])
  userEvent.click(screen.getByText("Release"))
  await waitFor(() => screen.getByText("Sending release."))
  await waitFor(() => screen.getByText(/Release Result/))
  expect(screen.getByText((/JSON Request/))).toBeTruthy()
  expect(screen.getByText("XML Request")).toBeTruthy()
  expect(screen.getByText((/JSON Response/))).toBeTruthy()
  expect(screen.getByText("XML Response")).toBeTruthy()
  expect(pretty(container.innerHTML)).toMatchSnapshot()
})

test("Displays release error response", async () => {
  const withDispenser: DispenserDetails = {
    odsCode: "testOdsCode",
    name: "testName",
    tel: "00000"
  }
  mock.onAny(releaseUrl).reply(200, {
    prescriptionIds: [],
    withDispenser,
    success: false,
    request: {req: "JSON Request"},
    request_xml: "XML Request",
    response: {res: "JSON Response"},
    response_xml: "XML Response"
  })

  const container = await renderPage()
  const pharmacyContainer = await screen.findByLabelText<HTMLElement>("Pharmacy to release prescriptions to")
  const pharmacyRadios = pharmacyContainer.getElementsByClassName("nhsuk-radios__input") as HTMLCollectionOf<HTMLInputElement>
  userEvent.click(pharmacyRadios[0])
  userEvent.click(screen.getByText("Release"))
  await waitFor(() => screen.getByText("Sending release."))
  await waitFor(() => screen.getByText(/Release Result/))
  expect(screen.getByText(/testOdsCode/)).toBeTruthy()
  expect(screen.getByText(/00000/)).toBeTruthy()
  expect(pretty(container.innerHTML)).toMatchSnapshot()
})

function createReleaseFormValuesCustom(pharmacyIdentifierCustom: string): ReleaseFormValues{
  return {releaseType: "all",
    pharmacy: "custom",
    customPharmacy: pharmacyIdentifierCustom
  }
}

describe("multiple instructions", () => {
  test("Prescription is released to the pharmacy with the code VNFKT", () => {
    const input: ReleaseFormValues = {
      releaseType: "all",
      pharmacy: "VNFKT"
    }
    const auth = "User"
    const result = createRelease(input, auth)
    const organization = result.parameter[0].resource as fhir.Organization
    const identifierValue = organization.identifier[0].value
    expect(identifierValue).toBe("VNFKT")
  })

  test("Prescription is released to the pharmacy with the code YGM1E", () => {
    const input: ReleaseFormValues = {
      releaseType: "all",
      pharmacy: "YGM1E"
    }
    const auth = "User"
    const result = createRelease(input, auth)
    const organization = result.parameter[0].resource as fhir.Organization
    const identifierValue = organization.identifier[0].value
    expect(identifierValue).toBe("YGM1E")
  })

  test("Prescription is released to pharmacy with a custom ODS code", () => {
    const input = createReleaseFormValuesCustom("FCG71")
    const auth = "User"
    const result = createRelease(input, auth)
    const organization = result.parameter[0].resource as fhir.Organization
    const identifierValue = organization.identifier[0].value
    expect(identifierValue).toBe("FCG71")
  })
})

async function renderPage() {
  const {container} = renderWithContext(<BrowserRouter><ReleasePage prescriptionId={prescriptionId}/></BrowserRouter>, context)
  await waitFor(() => screen.getByText("Release prescription(s)"))
  return container
}
