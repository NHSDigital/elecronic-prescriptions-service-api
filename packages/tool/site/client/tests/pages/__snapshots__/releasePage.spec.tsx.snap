// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Displays release error response 1`] = `
"<h1 class="nhsuk-label-wrapper"><label class="nhsuk-label nhsuk-label--xl">Release Result <svg class="nhsuk-icon nhsuk-icon__cross" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" aria-hidden="true" height="34" width="34">
      <path d="M17 18.5c-.4 0-.8-.1-1.1-.4l-10-10c-.6-.6-.6-1.6 0-2.1.6-.6 1.5-.6 2.1 0l10 10c.6.6.6 1.5 0 2.1-.3.3-.6.4-1 .4z" fill="#d5281b"></path>
      <path d="M7 18.5c-.4 0-.8-.1-1.1-.4-.6-.6-.6-1.5 0-2.1l10-10c.6-.6 1.5-.6 2.1 0 .6.6.6 1.5 0 2.1l-10 10c-.3.3-.6.4-1 .4z" fill="#d5281b"></path>
    </svg></label></h1>
<p>Prescription has been released by testName - testOdsCode.</p>
<p>Tel: 00000.</p>
<div class="nhsuk-action-link"><a class="nhsuk-action-link__link" href="baseUrl/prescribe/cancel?prescription_id=7A9089-A83008-56A03J"><svg class="nhsuk-icon nhsuk-icon__arrow-right-circle" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" aria-hidden="true" height="36" width="36">
      <path d="M0 0h24v24H0z" fill="none"></path>
      <path d="M12 2a10 10 0 0 0-9.95 9h11.64L9.74 7.05a1 1 0 0 1 1.41-1.41l5.66 5.65a1 1 0 0 1 0 1.42l-5.66 5.65a1 1 0 0 1-1.41 0 1 1 0 0 1 0-1.41L13.69 13H2.05A10 10 0 1 0 12 2z"></path>
    </svg><span class="nhsuk-action-link__text">Cancel prescription</span></a></div>
<div class="nhsuk-action-link"><a class="nhsuk-action-link__link" href="baseUrl/view?prescription_id=7A9089-A83008-56A03J"><svg class="nhsuk-icon nhsuk-icon__arrow-right-circle" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" aria-hidden="true" height="36" width="36">
      <path d="M0 0h24v24H0z" fill="none"></path>
      <path d="M12 2a10 10 0 0 0-9.95 9h11.64L9.74 7.05a1 1 0 0 1 1.41-1.41l5.66 5.65a1 1 0 0 1 0 1.42l-5.66 5.65a1 1 0 0 1-1.41 0 1 1 0 0 1 0-1.41L13.69 13H2.05A10 10 0 1 0 12 2z"></path>
    </svg><span class="nhsuk-action-link__text">Check prescription status</span></a></div>
<div class="nhsuk-expander-group">
  <details class="nhsuk-details nhsuk-expander">
    <summary class="nhsuk-details__summary"><span class="nhsuk-details__summary-text">1 - FHIR Request that a supplier would send to the FHIR API</span></summary>
    <div class="nhsuk-details__text">
      <div class="sc-eqUzNf chXQhu"><button class="nhsuk-button sc-iGgVNO cEsNDy" aria-disabled="false" type="submit">Copy</button><a class="nhsuk-button sc-iGgVNO cEsNDy" role="button" aria-disabled="false" draggable="false" href="data:application/json;charset=utf-8,%7B%0A%20%20%22req%22%3A%20%22JSON%20Request%22%0A%7D" download="message">Download</a></div>
      <div class="_2IvMF _GzYRV">
        <div class="_2bkNM" role="list"><span class="_oLqym _f10Tu _1MFti _1LId0" role="button" tabindex="0" aria-label="collapse JSON" aria-expanded="true" aria-controls=":jsnvw:3:"></span><span class="_3uHL6 _3eOF8">{</span>
          <div id=":jsnvw:3:">
            <div class="_2bkNM" role="listitem"><span class="_1MGIk">req:</span><span class="_vGjyY">"JSON Request"</span></div>
          </div><span class="_3uHL6 _3eOF8">}</span>
        </div>
      </div>
    </div>
  </details>
  <details class="nhsuk-details nhsuk-expander">
    <summary class="nhsuk-details__summary"><span class="nhsuk-details__summary-text">2 - Equivalent hl7v3 that the FHIR API has sent to spine</span></summary>
    <div class="nhsuk-details__text">
      <div class="sc-eqUzNf chXQhu"><button class="nhsuk-button sc-iGgVNO cEsNDy" aria-disabled="false" type="submit">Copy</button><a class="nhsuk-button sc-iGgVNO cEsNDy" role="button" aria-disabled="false" draggable="false" href="data:text/xml;charset=utf-8,XML%20Request" download="message">Download</a></div><pre class="sc-dcJtft jyMVel">XML Request</pre>
    </div>
  </details>
  <details class="nhsuk-details nhsuk-expander">
    <summary class="nhsuk-details__summary"><span class="nhsuk-details__summary-text">3 -  The hl7v3 spine response that the FHIR API received from spine</span></summary>
    <div class="nhsuk-details__text">
      <div class="sc-eqUzNf chXQhu"><button class="nhsuk-button sc-iGgVNO cEsNDy" aria-disabled="false" type="submit">Copy</button><a class="nhsuk-button sc-iGgVNO cEsNDy" role="button" aria-disabled="false" draggable="false" href="data:text/xml;charset=utf-8,XML%20Response" download="message">Download</a></div><pre class="sc-dcJtft jyMVel">XML Response</pre>
    </div>
  </details>
  <details class="nhsuk-details nhsuk-expander">
    <summary class="nhsuk-details__summary"><span class="nhsuk-details__summary-text">4 - FHIR Response that a supplier will receive from the FHIR API</span></summary>
    <div class="nhsuk-details__text">
      <div class="sc-eqUzNf chXQhu"><button class="nhsuk-button sc-iGgVNO cEsNDy" aria-disabled="false" type="submit">Copy</button><a class="nhsuk-button sc-iGgVNO cEsNDy" role="button" aria-disabled="false" draggable="false" href="data:application/json;charset=utf-8,%7B%0A%20%20%22res%22%3A%20%22JSON%20Response%22%0A%7D" download="message">Download</a></div>
      <div class="_2IvMF _GzYRV">
        <div class="_2bkNM" role="list"><span class="_oLqym _f10Tu _1MFti _1LId0" role="button" tabindex="0" aria-label="collapse JSON" aria-expanded="true" aria-controls=":jsnvw:4:"></span><span class="_3uHL6 _3eOF8">{</span>
          <div id=":jsnvw:4:">
            <div class="_2bkNM" role="listitem"><span class="_1MGIk">res:</span><span class="_vGjyY">"JSON Response"</span></div>
          </div><span class="_3uHL6 _3eOF8">}</span>
        </div>
      </div>
    </div>
  </details>
</div>
<div class="sc-eqUzNf chXQhu"><button class="nhsuk-button nhsuk-button--secondary" aria-disabled="false" type="button">Back</button></div>"
`;

exports[`Displays release form 1`] = `
"<h1 class="nhsuk-label-wrapper"><label class="nhsuk-label nhsuk-label--xl">Release prescription(s)</label></h1>
<form>
  <div class="nhsuk-form-group">
    <fieldset class="nhsuk-fieldset">
      <div class="nhsuk-form-group"><label class="nhsuk-label nhsuk-label--s" id="releaseType--label" for="releaseType">Choose how you want to release prescription(s)</label>
        <div class="nhsuk-radios" id="releaseType" aria-labelledby="releaseType--label" value="prescriptionId">
          <div class="nhsuk-radios__item"><input class="nhsuk-radios__input" id="releaseType-1" name="releaseType" aria-labelledby="releaseType-1--label" type="radio" value="all"><label class="nhsuk-label nhsuk-radios__label" id="releaseType-1--label" for="releaseType-1">All nominated prescriptions for the below pharmacy</label></div>
          <div class="nhsuk-radios__item"><input class="nhsuk-radios__input" id="releaseType-2" name="releaseType" aria-labelledby="releaseType-2--label" type="radio" value="prescriptionId" checked=""><label class="nhsuk-label nhsuk-radios__label" id="releaseType-2--label" for="releaseType-2">A single prescription by ID</label></div>
          <div class="nhsuk-radios__item"><input class="nhsuk-radios__input" id="releaseType-3" name="releaseType" aria-labelledby="releaseType-3--label" type="radio" value="custom"><label class="nhsuk-label nhsuk-radios__label" id="releaseType-3--label" for="releaseType-3">With a FHIR release message</label></div>
        </div>
      </div>
      <div class="nhsuk-form-group"><label class="nhsuk-label" id="prescriptionId--label" for="prescriptionId">Prescription ID</label><input class="nhsuk-input nhsuk-input--width-30" type="text" aria-labelledby="prescriptionId--label" name="prescriptionId" id="prescriptionId" value="7A9089-A83008-56A03J"></div>
      <div class="nhsuk-form-group"><label class="nhsuk-label nhsuk-label--s" id="pharmacy--label" for="pharmacy">Pharmacy to release prescriptions to</label>
        <div class="nhsuk-radios" id="pharmacy" aria-labelledby="pharmacy--label" value="">
          <div class="nhsuk-radios__item"><input class="nhsuk-radios__input" id="pharmacy-1" name="pharmacy" aria-labelledby="pharmacy-1--label" type="radio" value="VNFKT"><label class="nhsuk-label nhsuk-radios__label" id="pharmacy-1--label" for="pharmacy-1">VNFKT - FIVE STAR HOMECARE LEEDS LTD</label></div>
          <div class="nhsuk-radios__item"><input class="nhsuk-radios__input" id="pharmacy-2" name="pharmacy" aria-labelledby="pharmacy-2--label" type="radio" value="YGM1E"><label class="nhsuk-label nhsuk-radios__label" id="pharmacy-2--label" for="pharmacy-2">YGM1E - MBBM HEALTHCARE TECHNOLOGIES LTD</label></div>
          <div class="nhsuk-radios__item"><input class="nhsuk-radios__input" id="pharmacy-3" name="pharmacy" aria-labelledby="pharmacy-3--label" type="radio" value="custom"><label class="nhsuk-label nhsuk-radios__label" id="pharmacy-3--label" for="pharmacy-3">Other</label></div>
        </div>
      </div>
    </fieldset>
  </div>
  <div class="sc-eqUzNf chXQhu"><button class="nhsuk-button" aria-disabled="false" type="submit">Release</button><button class="nhsuk-button nhsuk-button--secondary" aria-disabled="false" type="button">Back</button></div>
</form>"
`;

exports[`Displays release result 1`] = `
"<h1 class="nhsuk-label-wrapper"><label class="nhsuk-label nhsuk-label--xl">Release Result <svg class="nhsuk-icon nhsuk-icon__tick" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" aria-hidden="true" height="34" width="34">
      <path stroke-width="4" stroke-linecap="round" stroke="#007f3b" fill="none" d="M18.4 7.8l-8.5 8.4L5.6 12"></path>
    </svg></label></h1>
<div class="nhsuk-expander-group">
  <details class="nhsuk-details nhsuk-expander">
    <summary class="nhsuk-details__summary"><span class="nhsuk-details__summary-text">1 - FHIR Request that a supplier would send to the FHIR API</span></summary>
    <div class="nhsuk-details__text">
      <div class="sc-eqUzNf chXQhu"><button class="nhsuk-button sc-iGgVNO cEsNDy" aria-disabled="false" type="submit">Copy</button><a class="nhsuk-button sc-iGgVNO cEsNDy" role="button" aria-disabled="false" draggable="false" href="data:application/json;charset=utf-8,%7B%0A%20%20%22req%22%3A%20%22JSON%20Request%22%0A%7D" download="message">Download</a></div>
      <div class="_2IvMF _GzYRV">
        <div class="_2bkNM" role="list"><span class="_oLqym _f10Tu _1MFti _1LId0" role="button" tabindex="0" aria-label="collapse JSON" aria-expanded="true" aria-controls=":jsnvw:1:"></span><span class="_3uHL6 _3eOF8">{</span>
          <div id=":jsnvw:1:">
            <div class="_2bkNM" role="listitem"><span class="_1MGIk">req:</span><span class="_vGjyY">"JSON Request"</span></div>
          </div><span class="_3uHL6 _3eOF8">}</span>
        </div>
      </div>
    </div>
  </details>
  <details class="nhsuk-details nhsuk-expander">
    <summary class="nhsuk-details__summary"><span class="nhsuk-details__summary-text">2 - Equivalent hl7v3 that the FHIR API has sent to spine</span></summary>
    <div class="nhsuk-details__text">
      <div class="sc-eqUzNf chXQhu"><button class="nhsuk-button sc-iGgVNO cEsNDy" aria-disabled="false" type="submit">Copy</button><a class="nhsuk-button sc-iGgVNO cEsNDy" role="button" aria-disabled="false" draggable="false" href="data:text/xml;charset=utf-8,XML%20Request" download="message">Download</a></div><pre class="sc-dcJtft jyMVel">XML Request</pre>
    </div>
  </details>
  <details class="nhsuk-details nhsuk-expander">
    <summary class="nhsuk-details__summary"><span class="nhsuk-details__summary-text">3 -  The hl7v3 spine response that the FHIR API received from spine</span></summary>
    <div class="nhsuk-details__text">
      <div class="sc-eqUzNf chXQhu"><button class="nhsuk-button sc-iGgVNO cEsNDy" aria-disabled="false" type="submit">Copy</button><a class="nhsuk-button sc-iGgVNO cEsNDy" role="button" aria-disabled="false" draggable="false" href="data:text/xml;charset=utf-8,XML%20Response" download="message">Download</a></div><pre class="sc-dcJtft jyMVel">XML Response</pre>
    </div>
  </details>
  <details class="nhsuk-details nhsuk-expander">
    <summary class="nhsuk-details__summary"><span class="nhsuk-details__summary-text">4 - FHIR Response that a supplier will receive from the FHIR API</span></summary>
    <div class="nhsuk-details__text">
      <div class="sc-eqUzNf chXQhu"><button class="nhsuk-button sc-iGgVNO cEsNDy" aria-disabled="false" type="submit">Copy</button><a class="nhsuk-button sc-iGgVNO cEsNDy" role="button" aria-disabled="false" draggable="false" href="data:application/json;charset=utf-8,%7B%0A%20%20%22res%22%3A%20%22JSON%20Response%22%0A%7D" download="message">Download</a></div>
      <div class="_2IvMF _GzYRV">
        <div class="_2bkNM" role="list"><span class="_oLqym _f10Tu _1MFti _1LId0" role="button" tabindex="0" aria-label="collapse JSON" aria-expanded="true" aria-controls=":jsnvw:2:"></span><span class="_3uHL6 _3eOF8">{</span>
          <div id=":jsnvw:2:">
            <div class="_2bkNM" role="listitem"><span class="_1MGIk">res:</span><span class="_vGjyY">"JSON Response"</span></div>
          </div><span class="_3uHL6 _3eOF8">}</span>
        </div>
      </div>
    </div>
  </details>
</div>
<div class="sc-eqUzNf chXQhu"><button class="nhsuk-button nhsuk-button--secondary" aria-disabled="false" type="button">Back</button></div>"
`;
